library/bigip_ssl_certificate.py

```diff
@@ -0,0 +1,449 @@
+        description:
+            - Required if the format is PEM.
+            - This is the filename of the certificate.
+        required: False
+        default: none
+    key_pem_file:
+        description:
+            - Require if the format is PEM.
+            - This is the filename of the private key,
+        required: False
+        default: none
+    pkcs12_file:
+        description:
+            - Current unused
+        required: False
+        default: none
+    pkcs12_password:
+        description:
+            - Currently unused
+        required: False
+        default: none
+    validata_cert:
+        description:
+            - Validate the certificate on the remote Big-IP is valid.
+            - Currently f5-sdk appears to hardcode this to False
+        required: False
+        default: True
+        choices: [true. false]
+notes:
+    - bigsuds required for icontrol connection
+    - f5-sdk version 0.1.7 required for rest connection
+    - PKCS12 will be supported when f5-sdk adds support.
+requirements:
+    - bigsuds
+    - f5-sdk
+author:
+    - Kevin Coming <kevcom@gmail.com>
+
+'''
+
+EXAMPLES = '''
+    - name: "Import PEM Certificate"
+      local_action:
+        name: certificate-name
+        module: bigip_ssl_certificate
+        server: bigip-addr
+        username: username
+        password: password
+        connection: rest
+        state: present
+        cert_format: pem
+        cert_pem_file: /path/to/cert.crt
+        key_pem_file: /path/to/key.key
+
+    - name: "Delete Certificate"
+      local_action:
+        name: certificate-name
+        module: bigip_ssl_certificate
+        server: bigip-addr
+        username: username
+        password: password
+        state: absent
+        connection: rest
+
+'''
+
+
+try:
+    from OpenSSL import crypto
+
+except:
+    found_OpenSSL = False
+else:
+    found_OpenSSL = True
+
+try:
+    import bigsuds
+except:
+    bigsuds_found = False
+else:
+    bigsuds_found = True
+
+try:
+    from f5.bigip import ManagementRoot
+    from icontrol.session import iControlUnexpectedHTTPError
+except:
+    f5sdk_found = False
+else:
+    f5sdk_found = True
+
+
+def read_file(filename):
+    with open(filename) as f:
+        data = f.read()
+    return data
+
+
+def pkcs12_to_pem(pkcs12data, password=''):
+    pkcs12 = crypto.load_pkcs12(pkcs12data, password)
+    key = crypto.dump_privatekey(crypto.FILETYPE_PEM, pkcs12.get_privatekey())
+    cert = crypto.dump_certificate(crypto.FILETYPE_PEM,
+                                   pkcs12.get_certificate())
+    return (cert, key)
+
+
+class F5ModuleError(Exception):
+    pass
+
+
+class BigIPCommon:
+    def __init__(self, hostname, username, password, verify=True):
+        self.hostname = hostname
+        self.username = username
+        self.password = password
+        self.verify = verify
+        self.api = self.connection(hostname, username, password, verify)
+
+
+class BigIPiControl(BigIPCommon):
```

Comment:
If this module can work entirely with REST, I recommend we remove the SOAP functionality in the interest of driving REST adoption.